crates/ide/src/references/rename.rs

```diff
@@ -313,11 +366,36 @@ mod tests {
     }
 
     #[test]
-    fn test_rename_to_invalid_identifier() {
+    fn test_rename_to_invalid_identifier1() {
         let (analysis, position) = analysis_and_position(r#"fn main() { let i<|> = 1; }"#);
         let new_name = "invalid!";
         let source_change = analysis.rename(position, new_name).unwrap();
-        assert!(source_change.is_none());
+        assert_eq!(
+            "Invalid name 'invalid!': not an identifier",
+            source_change.err().unwrap().to_string()
+        );
+    }
+
+    #[test]
+    fn test_rename_to_invalid_identifier2() {
+        let (analysis, position) = analysis_and_position(r#"fn main() { let i<|> = 1; }"#);
+        let new_name = "multiple tokens";
+        let source_change = analysis.rename(position, new_name).unwrap();
+        assert_eq!(
+            "Invalid name 'multiple tokens': not an identifier",
+            source_change.err().unwrap().to_string()
+        );
+    }
+
+    #[test]
+    fn test_rename_to_invalid_identifier3() {
+        let (analysis, position) = analysis_and_position(r#"fn main() { let i<|> = 1; }"#);
+        let new_name = "let";
+        let source_change = analysis.rename(position, new_name).unwrap();
+        assert_eq!(
+            "Invalid name 'let': not an identifier",
```

Comment:
```suggestion
            "Invalid name `let`: not an identifier",
```

we use backticks to surround code.